lazyflow/operators/opResize.py

```diff
@@ -0,0 +1,253 @@
+        NEAREST = 0
+        LINEAR = 1
+
+    semantics_to_interpolation = {
+        ImageTypes.Intensities: Interpolation.LINEAR,
+        ImageTypes.Labels: Interpolation.NEAREST,
+    }
+
+    required_min_padding = {  # minimum pixels of halo required for accurate interpolation
+        Interpolation.NEAREST: 0,
+        Interpolation.LINEAR: 2,
+    }
+
+    RawImage = InputSlot()
+    TargetShape = InputSlot()
+    InterpolationOrder = InputSlot(value=Interpolation.LINEAR)
+    ResizedImage = OutputSlot()
+
+    def __init__(
+        self,
+        parent=None,
+        graph=None,
+        RawImage: Slot = None,
+        TargetShape: Union[Tuple[int, ...], Slot, None] = None,
+        InterpolationOrder: Union[Interpolation, Slot, None] = None,
+        **kwargs,
+    ):
+        super().__init__(parent=parent, graph=graph, **kwargs)
+        self.scaling_factors: Union[NDArray, None] = None  # one factor per axis. Factor 2.0 = downscale to half size
+        self.antialiasing_sigmas: Union[NDArray, None] = None  # one sigma per axis
+        self.RawImage.setOrConnectIfAvailable(RawImage)
+        self.TargetShape.setOrConnectIfAvailable(TargetShape)
+        self.InterpolationOrder.setOrConnectIfAvailable(InterpolationOrder)
+
+    def setupOutputs(self):
+        if self.TargetShape.value == self.RawImage.meta.shape:
+            # Shortcut: no resizing needed
+            self.ResizedImage.connect(self.RawImage)
+        input_axiskeys = self.RawImage.meta.getAxisKeys()
+        assert len(input_axiskeys) == len(
+            self.TargetShape.value
+        ), f"Input image ({self.RawImage.meta.shape}) and target shape ({self.TargetShape.value}) must have same axes"
+        assert (
+            "c" not in input_axiskeys
+            or self.TargetShape.value[input_axiskeys.index("c")] == self.RawImage.meta.getTaggedShape()["c"]
+        ), "Cannot resize along channel axis"
+        if (
+            "t" in input_axiskeys
+            and self.TargetShape.value[input_axiskeys.index("t")] != self.RawImage.meta.getTaggedShape()["t"]
+        ):
+            logger.warning("Resizing along time axis. Are you sure this is what you want?")
+        self.ResizedImage.meta.assignFrom(self.RawImage.meta)
+        self.ResizedImage.meta.shape = self.TargetShape.value
+
+        # Scaling factors are the inverse of zoom factors in the scipy.ndimage sense.
+        # E.g. 2 if the output shape is half the input shape (rather than 0.5).
+        shape_in = self.RawImage.meta.shape
+        shape_out = self.ResizedImage.meta.shape
+        self.scaling_factors = np.divide(shape_in, shape_out)
+
+        interpolation_order = self.InterpolationOrder.value
+        # If higher-order interpolation is desired, need to figure out required padding
+        assert (
+            interpolation_order in self.required_min_padding
+        ), "supports only order 0 (nearest-neighbor) or 1 (linear)"
+
+        if interpolation_order > 0:
```

Comment:
hard coded enum constants as strings here. Potential source for future inconsistency.

```suggestion
        if interpolation_order != self.Interpolation.NEAREST :
```